### PR TITLE
feat: add ensureSymlink/ensureSymlinkSync for fs modules

### PR DESCRIPTION
**Pull request by axetroy**
part of #261

not implement in Windows now.

### CODE REVIEW

File: fs/ensure_syslink_test.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    ensureSyslinkSync(testFile, path.join(testDir, "test1.txt"));
+  });
+
+  assertThrows(() => {
+    Deno.statSync(testFile);
+    throw new Error("test file should exists.");
+  });
+});
+
+test(async function ensureSyslinkIfItExist() {
```
Comment (zekth) [inline fs/ensure_syslink_test.ts]: add condition for Deno.platform == 'win' and add a assertThrow() to handle the no implemented.

Comment (axetroy) [reply fs/ensure_syslink_test.ts]: make sense.

**Comment by axetroy** [comment]
> I don't think Should skip is a good solution. I'd test assertThrows(ensureSymlink()) if deno.platform === 'win'

or should wait for support `windows`?

**Comment by zekth** [comment]
You may have to do a conditionnal if(win) in the tests due to the notimplemend status. No? Seems legit to me.

**Comment by axetroy** [comment]
@zekth Yes. you are right.

I found that `denolan/deno` has many places to write like this, I should change it.

**Pull request merged**